src/beignet/_bisect.py

```diff
@@ -0,0 +1,59 @@
+from dataclasses import dataclass
+from typing import Callable, Literal
+
+import torch
+from torch import Tensor
+
+from beignet._root_scalar import RootSolutionInfo
+
+
+def bisect(
+    f: Callable,
+    *args,
+    lower: float,
+    upper: float,
+    rtol: float | None = None,
+    atol: float | None = None,
+    maxiter: int = 100,
+    return_solution_info: bool = False,
+    dtype=None,
+    device=None,
+    **_,
+) -> Tensor | tuple[Tensor, RootSolutionInfo]:
+    a = torch.tensor(lower, dtype=dtype, device=device)
+    b = torch.tensor(upper, dtype=dtype, device=device)
+
+    fa = f(a, *args)
+    fb = f(b, *args)
+    c = (a + b) / 2
+    fc = f(c, *args)
+
+    eps = torch.finfo(fa.dtype).eps
```

Comment:
`epsilon = torch.finfo(dtype).eps` right?

Reply:
it's possible for the output of func to have a different dtype so this seemed safer